packages/examples/cvat/exchange-oracle/src/services/cloud/types.py

```diff
@@ -31,24 +30,22 @@ def from_str(cls, provider: str) -> CloudProviders:
 
 
 class BucketCredentials:
-    def to_dict(self) -> Dict:
+    def to_dict(self) -> dict:
         if not is_dataclass(self):
             raise NotImplementedError
 
         return asdict(self)
 
     @classmethod
-    def from_storage_config(cls, config: Type[StorageConfig]) -> Optional[BucketCredentials]:
+    def from_storage_config(cls, config: type[StorageConfig]) -> BucketCredentials | None:
         credentials = None
 
         if (config.access_key or config.secret_key) and config.provider.lower() != "aws":
             raise ValueError(
                 "Invalid storage configuration. The access_key/secret_key pair"
                 f"cannot be specified with {config.provider} provider"
             )
-        elif (
-            bool(config.access_key) ^ bool(config.secret_key)
-        ) and config.provider.lower() == "aws":
+        if (bool(config.access_key) ^ bool(config.secret_key)) and config.provider.lower() == "aws":
```

Comment:
This transition makes it look like it is else, while it's not. I suggest to add 1 empty line between the blocks to visually separate them.

Reply:
Thanks for the suggestion. I've added an extra line to improve the visual separation.

> One more downside I can see here is that now once you decide to change logic in this condition, you might need to change it back to `elif` everywhere.

While that’s a valid point, I believe it's a small price to pay for the gain in clarity and readability.

Initially, I had the same reservations about following this rule of avoiding unnecessary `elif` statements, but after working with it for a while, I've found it to be more beneficial. 

Using unnecessary constructs like `elif` in combination with early `return` or `raise` can lead to confusion. It may prompt the reader to search for a scenario where this construct would make a difference, only to find that there isn't one. This wastes time and mental energy, which could be better spent understanding the actual logic of the code.